packages/jaeger-ui/src/model/ddg/transform-ddg-data.tsx

```diff
@@ -78,6 +79,7 @@ export default function transformDdgData(
     return path;
   });
 
+  // Assign visibility indices such there is a positive, dependent correlation between visibilityIdx and distance
```

Comment:
That sounds reasonable. I implemented the sort as:
```
const stringifyPayloadEntry = ({ service, operation }: TDdgPayloadEntry) => `${service}::${operation}`;

/* ... */
  const paths = payload
    .slice()
    .sort((a, b) =>
      a
        .map(stringifyPayloadEntry)
        .join()
        .localeCompare(b.map(stringifyPayloadEntry).join())
    )
```
It seems to work well and should work at scale. Did you have a different sort in mind?

Reply:
Not so sure I'd say this will scale, localeCompare is dog slow:

![Screen Shot 2019-05-22 at 1 55 11 PM](https://user-images.githubusercontent.com/2304337/58196915-46bad280-7c99-11e9-9b33-bb6c4e56a925.png)

https://jsperf.com/localecompare/38

Reply:
Wow that is really slow! I had no idea. I will update it.
I think for our use of static ordering, not necessarily the most logical ordering (e.g.: `aZ` before `aa` is fine as long as it is consistent), `(a, b) => a > b ? 1 : -1` should be a sufficient comparison to avoid needing to convert to lowercase unnecessarily.